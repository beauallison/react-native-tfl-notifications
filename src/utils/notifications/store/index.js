import { FileSystem } from 'expo';
import { LINES } from '../../../../constants';

const path = `${FileSystem.documentDirectory}settings.json`;

module.exports.exists = () =>
  FileSystem.getInfoAsync(path)
    .then(({ exists }) => exists);

module.exports.setup = async () => {
  const stations = {};
  Object.values(LINES).forEach((station) => {
    stations[station] = false;
  });

  await exports.store(stations);
};

module.exports.update = async (ID, value) => {
  const stations = await exports.get();
  stations[ID] = value;
  await exports.store(stations);
};

module.exports.get = async () => {
  const store = await FileSystem.readAsStringAsync(path);

  return JSON.parse(store);
};

module.exports.store = async (stations) => {
  const contents = JSON.stringify(stations);

  await FileSystem.writeAsStringAsync(path, contents);
};
