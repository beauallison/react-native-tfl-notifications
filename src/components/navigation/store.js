import _ from 'lodash';
import { LINES } from '../../../constants';
import { store, update } from '../../utils/notifications';

let Store = {};

(async () => {
  const exists = await store.exists();
  if (!exists) { await store.setup(); }
  Store = await store.get();
})();

const updateStation = (station, status) => {
  Store[station] = status;
  return status;
};

export default _.map(LINES, (station, ID) => ({
  ID,
  station,
  enabled: Store[station],
  updateStation: input => updateStation(station, input),
}));

export const save = () => update(Store);
