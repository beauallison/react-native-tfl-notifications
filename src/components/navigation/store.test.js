import store, { save } from './store';

jest.mock('expo', () => require('../../utils/notifications/store/_mocks_/expo')); // eslint-disable-line global-require

describe('components/navigation/store', () => {
  it('should return the notifications store', () =>
    expect(store).toMatchSnapshot());

  it('should update a station', () =>
    expect(store[0].updateStation(true)).toBeTruthy());

  it('should save user settings', async () => {
    fetch.mockResponseOnce('OK');

    const output = await save();
    expect(output).toEqual('OK');
  });
});
