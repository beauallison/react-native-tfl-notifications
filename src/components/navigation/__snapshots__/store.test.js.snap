// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/navigation/store should return the notifications store 1`] = `
Array [
  Object {
    "ID": "BAKERLOO",
    "enabled": undefined,
    "station": "bakerloo",
    "updateStation": [Function],
  },
  Object {
    "ID": "CENTRAL",
    "enabled": undefined,
    "station": "central",
    "updateStation": [Function],
  },
  Object {
    "ID": "DISTRICT",
    "enabled": undefined,
    "station": "district",
    "updateStation": [Function],
  },
  Object {
    "ID": "CIRCLE",
    "enabled": undefined,
    "station": "circle",
    "updateStation": [Function],
  },
  Object {
    "ID": "HAMMERSMITH",
    "enabled": undefined,
    "station": "hammersmith-city",
    "updateStation": [Function],
  },
  Object {
    "ID": "JUBILEE",
    "enabled": undefined,
    "station": "jubilee",
    "updateStation": [Function],
  },
  Object {
    "ID": "LONDON",
    "enabled": undefined,
    "station": "london-overground",
    "updateStation": [Function],
  },
  Object {
    "ID": "METROPOLITAN",
    "enabled": undefined,
    "station": "metropolitan",
    "updateStation": [Function],
  },
  Object {
    "ID": "NORTHERN",
    "enabled": undefined,
    "station": "northern",
    "updateStation": [Function],
  },
  Object {
    "ID": "PICCADILLY",
    "enabled": undefined,
    "station": "piccadilly",
    "updateStation": [Function],
  },
  Object {
    "ID": "VICTORIA",
    "enabled": undefined,
    "station": "victoria",
    "updateStation": [Function],
  },
  Object {
    "ID": "WATERLOO",
    "enabled": undefined,
    "station": "waterloo-city",
    "updateStation": [Function],
  },
]
`;
